aquia-inc/template-vite-react
# build(dev-deps): upgrade prettier, eslint, and related deps

## Summary

This PR upgrades several devDependencies related to eslint and prettier to their latest versions.

### Added

- None

### Changed

- Upgraded `eslint` from `^8.56.0` to `^8.57.0`
- Upgraded `eslint-plugin-jest` from `^27.8.0` to `^27.9.0`
- Upgraded `eslint-plugin-prettier` from `^5.1.3` to `^5.2.1`
- Upgraded `eslint-plugin-react` from `^7.33.2` to `^7.35.0`
- Upgraded `eslint-plugin-react-hooks` from `^4.6.0` to `^4.6.2`
- Upgraded `eslint-plugin-react-refresh` from `^0.4.8` to `^0.4.9`
- Upgraded `eslint-plugin-storybook` from `^0.6.15` to `^0.8.0`
- Upgraded `lint-staged` from `^14.0.1` to `^15.2.8`
- Upgraded `prettier` from `^3.2.5` to `^3.3.3`
- Upgraded `pretty-quick` from `^3.3.1` to `^4.0.0`
- Upgraded `typescript` from `^5.4.5` to `^5.5.4`

### Deprecated

- None

### Removed

- None

### Fixed

- None

## How to test

1. Checkout this branch: `git checkout origin/build/dependencies/6-eslint-prettier`
2. Run `yarn` and ensure all dependencies are correctly installed without duplication and that there are no issues related to missing or incompatible dependencies.
3. Run `yarn test` and verify that all tests run and pass.
4. Run `yarn dev` and verify that the application runs as expected.
5. Run `yarn build` and verify that the application builds as expected.
6. Run `yarn build-storybook` and verify the Storybook build succeeds.
